starknet_py/net/account/account_client.py

```diff
@@ -403,6 +407,63 @@ async def create_account(
             signer=signer,
         )
 
+    def sign_message(self, typed_data: TypedData) -> List[int]:
+        """
+        Sign an TypedData object for off-chain usage with the starknet private key and return the signature
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData object to be signed
+        :return: The signature of the TypedData object
+        """
+        return self.signer.sign_message(typed_data, self.address)
+
+    def hash_message(self, typed_data: TypedData) -> int:
+        """
+        Hash a TypedData object with pedersen hash and return the hash
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData object to be hashed
+        :return: the hash of the TypedData object
+        """
+        return get_message_hash(typed_data, self.address)
+
+    async def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
```

Comment:
I'd be more convenient for the user for the TypedData here to be typeddict and not dataclass. That way if the user has for example a json returned from somewhere it can be passed directly to this method. Currently they'd have to instantiate the dataclass themselves.

Reply:
Why wasn't this addessed